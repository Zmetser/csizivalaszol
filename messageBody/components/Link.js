/**
 * @flow
 * 
 * Create an anchor tag frm the Link structure
 */

import React from 'react'

import type { LinkNode } from '../types'

import Text from './Text'

type Props = LinkNode
export default function Link ({ value, href, target }: Props): React$Node<*> {
  const props = Object.assign({}, target ? { target } : null)
  return <a href={href} {...props}>{value.map(Text)}</a>
}
