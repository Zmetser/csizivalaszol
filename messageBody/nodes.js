/**
 * @flow
 *
 * All the usable nodes
 */

export type InlineStyle = 'Italic' | 'Bold' | 'Struck' | 'Small' | 'Code' | 'Subscript' | 'Superscript' | 'Underline' | 'Normal'

const TextNode = (value: string, styles: Array<InlineStyle>) => ({
  type: 'Text',
  value,
  styles
})

const LineBreak = () => ({ type: 'LineBreak' })

const Link = (value: Array<TextNode>, href: string, target?: ?string) => ({
  type: 'Link',
  value,
  href,
  target
})

export type InlineNode = TextNode | LineBreak | Link;

type ImageAttributes = { src: string, alt: string, width: number, height: number }
const Image = ({ src, alt, width, height }: ImageAttributes) => ({
  type: 'Image',
  src,
  alt,
  width,
  height
})

const Paragraph = (value: Array<InlineNode>) => ({
  type: 'Paragraph',
  value
})

module.exports = {
  TextNode,
  Paragraph,
  LineBreak,
  Link,
  Image
}
