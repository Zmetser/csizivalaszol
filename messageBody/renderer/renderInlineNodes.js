/**
 * @flow
 */

import * as React from 'react'
import { UnknownNodeError } from '../errors'

import type { InlineNode } from '../types'

import {
  Text,
  Link
} from '../components'

function createInlineNode (node: InlineNode, index): React.Element<*> {
  switch (node.type) {
    case 'Text': return Text(node, index)
    case 'Link': return <Link {...node} key={index} />
    default: throw new UnknownNodeError(node)
  }
}
export default function renderInlineNodes (nodes: Array<InlineNode>): Array<React.Element<*>> {
  return nodes.map(createInlineNode)
}
