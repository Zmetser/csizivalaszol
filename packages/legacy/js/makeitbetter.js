function del() {
  $elem = document.getElementById('page');
  $elem.value = "";
}