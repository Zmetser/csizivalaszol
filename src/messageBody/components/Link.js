/**
 * Create an anchor tag frm the Link structure
 */

import * as React from 'react'
import type { LinkNode } from '../types'
import Text from './Text'


type Props = { node: LinkNode }
export default function Link ({ node }: Props): React.Element<'a'> {
  const { value, href, target } = node
  const props = Object.assign({}, target ? { target } : null)
  return <a href={href} {...props}>{value.map((node, index) => <Text node={node} key={index} />)}</a>
}
