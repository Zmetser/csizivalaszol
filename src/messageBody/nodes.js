/**
 * All the usable nodes
 */

import type {
  ImageAttributes,
  ImageNode,
  InlineNode,
  InlineStyle,
  LineBreakNode,
  LinkNode,
  ParagraphNode,
  TextNode,
  ListType,
  ListItem,
  ListNode
} from './types'

export const createTextNode = (value: string, styles: Array<InlineStyle>): TextNode => ({
  type: 'Text',
  value,
  styles
})

export const createLineBreak = (): LineBreakNode => ({ type: 'LineBreak' })

export const createLink = (value: Array<TextNode>, href: string, target?: ?string): LinkNode => ({
  type: 'Link',
  value,
  href,
  target
})

export const createImage = ({ src, alt, width, height }: ImageAttributes): ImageNode => ({
  type: 'Image',
  src,
  alt,
  width,
  height
})

export const createParagraph = (value: Array<InlineNode>): ParagraphNode => ({
  type: 'Paragraph',
  value
})

export function createList<T: ListType> (items: Array<ListItem>, listType: T): ListNode<T> {
  return {
    type: 'List',
    listType,
    items
  }
}
