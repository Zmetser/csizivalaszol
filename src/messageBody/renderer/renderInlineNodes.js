/**
 * @flow
 */

import * as React from 'react'
import { UnknownNodeError } from '../errors'

import type { InlineNode } from '../types'

import {
  Text,
  Link,
  LineBreak
} from '../components'

function createInlineNode (node: InlineNode, index): React.Element<*> {
  switch (node.type) {
    case 'Text': return <Text node={node} key={index} />
    case 'Link': return <Link node={node} key={index} />
    case 'LineBreak': return <LineBreak key={index} />
    default: throw new UnknownNodeError(node)
  }
}
export default function renderInlineNodes (nodes: Array<InlineNode>): Array<React.Element<*>> {
  return nodes.map(createInlineNode)
}
