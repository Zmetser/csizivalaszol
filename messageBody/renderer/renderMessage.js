/**
 * @flow
 */

import * as React from 'react'
import { Paragraph } from '../components'

import type { BlockNode } from '../types.js'
import { UnknownNodeError } from '../errors'

export default function render (message: Array<BlockNode>): Array<React.Element<*>> {
  return message.map((node, index): React.Element<*> => {
    switch (node.type) {
      case 'Paragraph': return Paragraph({ value: node.value }, index)
      default: throw new UnknownNodeError(node)
    }
  })
}
