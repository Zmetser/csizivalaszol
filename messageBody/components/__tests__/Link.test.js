/**
 * @flow
 */
import renderer from 'react-test-renderer'

import Link from '../Link'

import { createTextNode } from '../../nodes.js'

const fooText = createTextNode('Foo', [])
const fooTextItalic = createTextNode('Foo', ['Italic'])

describe('<a />', () => {
  test('link', () => {
    const element = renderer.create(
      Link({ value: [fooText], href: '#' })
    )
    expect(element.toJSON()).toMatchSnapshot()
  })

  test('link with "_blank" target', () => {
    const element = renderer.create(
      Link({ value: [fooText], href: '#', target: '_blank' })
    )
    expect(element.toJSON()).toMatchSnapshot()
  })

  test('link with styled content', () => {
    const element = renderer.create(
      Link({ value: [fooTextItalic], href: '#', target: '_blank' })
    )
    expect(element.toJSON()).toMatchSnapshot()
  })

  test('link with a list of values', () => {
    const element = renderer.create(
      Link({ value: [fooTextItalic, fooTextItalic, fooTextItalic], href: '#', target: '_blank' })
    )
    expect(element.toJSON()).toMatchSnapshot()
  })
})
