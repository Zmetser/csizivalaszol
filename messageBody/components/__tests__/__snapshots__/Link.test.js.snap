// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<a /> link 1`] = `
<a
  href="#"
>
  Foo
</a>
`;

exports[`<a /> link with "_blank" target 1`] = `
<a
  href="#"
  target="_blank"
>
  Foo
</a>
`;

exports[`<a /> link with a list of values 1`] = `
<a
  href="#"
  target="_blank"
>
  <em>
    Foo
  </em>
  <em>
    Foo
  </em>
  <em>
    Foo
  </em>
</a>
`;

exports[`<a /> link with styled content 1`] = `
<a
  href="#"
  target="_blank"
>
  <em>
    Foo
  </em>
</a>
`;
